index.js

```diff
@@ -4,7 +4,10 @@ import getRawBody from 'raw-body'
 import assert from 'http-assert'
 import { validate } from './lib/validate.js'
 
-const handler = async (req, res, client, getCurrentRound) => {
+const handler = async (req, res, client, getCurrentRound, domain) => {
+  if (req.headers.host.split(':')[0] !== domain) {
```

Comment:
disregard the port, so that tests still work (where we don't know the port beforehand)